lib/sqlalchemy/cyextension/resultproxy.pyx

```diff
@@ -2,23 +2,18 @@
 # more improvements are likely possible
 import operator
 
-cdef int MD_INDEX = 0  # integer index in cursor.description
-cdef int _KEY_OBJECTS_ONLY = 1
-
-KEY_INTEGER_ONLY = 0
-KEY_OBJECTS_ONLY = _KEY_OBJECTS_ONLY
-
 cdef class BaseRow:
     cdef readonly object _parent
+    cdef readonly dict _key_to_index
     cdef readonly tuple _data
-    cdef readonly dict _keymap
-    cdef readonly int _key_style
 
-    def __init__(self, object parent, object processors, dict keymap, int key_style, object data):
+    def __init__(self, object parent, object processors, object data):
         """Row objects are constructed by CursorResult objects."""
 
         self._parent = parent
 
+        self._key_to_index = parent._key_to_index
```

Comment:
Maybe this dict could be passed from the outside as before to avoid a getattr for each row

Reply:
I think they would help 
<img width="1461" alt="Screenshot 2023-04-19 at 12 59 46 PM" src="https://user-images.githubusercontent.com/663432/233217579-79b66690-fc37-459b-ad94-dd03437e534b.png">

Reply:
We end up with

```diff
--- a/lib/sqlalchemy/cyextension/resultproxy.pyx
+++ b/lib/sqlalchemy/cyextension/resultproxy.pyx
@@ -7,12 +7,12 @@ cdef class BaseRow:
     cdef readonly dict _key_to_index
     cdef readonly tuple _data
 
-    def __init__(self, object parent, object processors, object data):
+    def __init__(self, object parent, object processors, dict key_to_index, object data):
         """Row objects are constructed by CursorResult objects."""
 
         self._parent = parent
 
-        self._key_to_index = parent._key_to_index
+        self._key_to_index = key_to_index
 
         if processors:
             self._data = tuple(
```

We get the `__Pyx_ArgTypeTest` check when passed in instead of `PyDict_CheckExact` but that should be neutral
<img width="1431" alt="Screenshot 2023-04-19 at 1 01 45 PM" src="https://user-images.githubusercontent.com/663432/233217856-fa791830-b5c7-439e-8844-5eae2f602e53.png">

but not getattr which is a win
<img width="485" alt="Screenshot 2023-04-19 at 1 01 52 PM" src="https://user-images.githubusercontent.com/663432/233217883-8981a0ca-0d5b-48d5-b47f-28fd2b9b36df.png">

Reply:
did that but it has no measurable impact